### ISSUE TITLE
More detailed usage information might be useful

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [peterbrittain/asciimatics#14]
So, this package looks of interest to non-Pythoneers, like myself. Unfortunately, `pip install asciimatics` isn't enough information if you don't already know quite a bit about Python:

1. For one thing, I wasn't sure whether I'd need to install Python 3 (as opposed to the Mac OS X default Python 2) to use this; (I'm only vaguely aware from the outside that there's a big split within the community, and I'm not sure what to do about it in this case?)
2. `pip install asciimatics` doesn't really tell me *where* it installs it. How do I run it? Did it install a binary somewhere (if so, it's not in my `$PATH`.) What does that binary *run*? Python files?
3. How, *specifically*, do I run examples from the `samples/` directory?

I read the [quick start guide](http://asciimatics.readthedocs.org/en/stable/intro.html#quick-start-guide), which really wasn't; and then ended up reading most of <https://packaging.python.org/en/latest/installing> before I discovered that I'm supposed to setup a `virtualenv`, and then `pip3 install -r requirements.txt` (or so I think? I'm not even sure if that ended up being correct!)

Even having done all of that, I can't run the samples: from every one I've tried, I get errors like the following:

```
(asciimatics)> python samples/xmas.py 
Traceback (most recent call last):
  File "samples/xmas.py", line 65, in <module>
    Screen.wrapper(demo)
  File "/Users/ec/Documents/Code/Source/asciimatics/lib/python3.5/site-packages/asciimatics-1.5.0-py3.5.egg/asciimatics/screen.py", line 479, in wrapper
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/curses/__init__.py", line 94, in wrapper
    return func(stdscr, *args, **kwds)
  File "/Users/ec/Documents/Code/Source/asciimatics/lib/python3.5/site-packages/asciimatics-1.5.0-py3.5.egg/asciimatics/screen.py", line 477, in _wrapper
  File "samples/xmas.py", line 61, in demo
    screen.play([Scene(effects, -1)], stop_on_resize=True)
  File "/Users/ec/Documents/Code/Source/asciimatics/lib/python3.5/site-packages/asciimatics-1.5.0-py3.5.egg/asciimatics/screen.py", line 672, in play
  File "/Users/ec/Documents/Code/Source/asciimatics/lib/python3.5/site-packages/asciimatics-1.5.0-py3.5.egg/asciimatics/screen.py", line 879, in clear
  File "/Users/ec/Documents/Code/Source/asciimatics/lib/python3.5/site-packages/asciimatics-1.5.0-py3.5.egg/asciimatics/screen.py", line 1445, in _change_colours
TypeError: write() argument must be str, not bytes
```

(… with the exception of `not_curses.py`, which gives me “ImportError: No module named 'blessed'.”)

I understand that explaining some of this may be beyond the scope of your own interests in this package. That's fine, and your perogative 😋; feel free to close this if so … but adding some more-newcomer-friendly instructions to the README might be worthwhile!

----
*(Unrelated side-note: I found you via being featured on the <asciinema.org> landing-page!)*

**Comment by peterbrittain** [comment]
@ELLIOTTCABLE Thanks for taking the time to tell me how you've struggled.  I'm sorry that it's been such hard work and will try to update the docs to make it easier for anyone new to Python.  To that end, let me take each point in turn...

* Asciimatics can run on Python 2 and 3 (v2.7, v3.3 and v3.4).  This is documented in the setup.py and hence [pypi](https://pypi.python.org/pypi/asciimatics/1.4.2).  I'll flag the broad compatibility (and reference a link for specifics) in the README.
* As a Python package, asciimatics installs to your site-packages (like every other package) when installed by pip.  It is not an application in its own right, but rather a library that allows you to write such applications.  This is why there are separate sample applications to show you what you can do.
* To run any of the samples, simply install asciimatics using pip and then download the sample and run it in the directory where you downloaded it by typing `python <sample name>` - e.g.`python fire.py`.  A simple way to download all the samples is to [clone the repository](https://help.github.com/articles/cloning-a-repository/). I'll flesh out the details of how to run them in the section of the README about the samples.
* The stack trace is a bug that was already reported and has just been fixed in v1.4.2.  Please let me know if you have hit any other problems with this version instead of your current version.
* The issue with not_curses.py is that it's actually a demonstration of how you could use blessed (a separate Python package) instead of the standard curses package.  To run it you therefore need to install blessed first (using pip).  I'll explain this (and how to run win_demo) better in the README section about the samples.

I don't think it's my place to document the entire Python back-story (e.g. Python 2 vs Python 3, standardization on pip as a package manager, etc) but as noted above I will try to provide some more pointers in the docs to address your issues.  If there's anything else beyond the above that you think would be useful, pleae let me know by replying to this issue and I'll try to incorporate it in the docs updates.

**Comment by ELLIOTTCABLE** [comment]
Ahhah!

So, turns out an *actual bug* was making me think I was completely missing how to use Python. That makes more sense.

If it's as simple as you describe, re: `pip install` and then `python file.py`, then that's probably all you need to document in the README, imho. I only thought it was more complicated because I kept running into unrelated walls. (=

**Comment by peterbrittain** [comment]
Ah.  If only I'd managed to fix that bug sooner...  OK - I'll still make most of the tweaks to the docs and then close this issue.

**Comment by peterbrittain** [comment]
Fixed in v1.5.0.
